core/storage/base_model/gae_models.py

```diff
@@ -542,17 +540,42 @@ def get_multi_versions(cls, entity_id, version_numbers):
         Returns:
             list(VersionedModel). Model instances representing the given
                 versions.
+
+        Raises:
+            ValueError. The given entity_id is invalid.
+            ValueError. Requested version number cannot be higher than the
+                current version number.
+            ValueError. At least one version number is invalid.
         """
         instances = []
-        try:
-            cls.get(entity_id)
-        except Exception as  e:
-            logging.error(e)
+
+        entity = cls.get(entity_id, strict=False)
+        if not entity:
+            raise ValueError('The given entity_id %s is invalid.' % (entity_id))
+        current_version = entity.version
+        if max(version_numbers) > current_version:
+            raise ValueError(
+                'Requested version number cannot be higher than the current '
+                'version number.')
+
+        snapshot_ids = []
         # pylint: disable=protected-access
         for version in version_numbers:
             snapshot_id = cls._get_snapshot_id(entity_id, version)
-            instances.append(cls(id=entity_id)._reconstitute_from_snapshot_id(
-                snapshot_id))
+            snapshot_ids.append(snapshot_id)
+
+        snapshot_models = cls.SNAPSHOT_CONTENT_CLASS.get_multi(snapshot_ids)
+        for index, snapshot_model in enumerate(snapshot_models):
```

Comment:
Why do you need index/enumerate? I don't think you use it.

Reply:
Removed it.